src/main/java/io/pinecone/clients/AsyncIndex.java

```diff
@@ -104,19 +104,75 @@ public ListenableFuture<QueryResponseWithUnsignedIndices> queryByVectorId(int to
         return query(topK, null, null, null, id, namespace, filter, false, false);
     }
 
+    @Override
+    public ListenableFuture<QueryResponseWithUnsignedIndices> queryByVectorId(int topK,
+                                                                              String id,
```

Comment:
Total nit, but maybe make `id` be `vectorId`, or make the method `querybyId`, just so the semantics of "id" carry through the method (or the semantics of "vector ID" if you go that route)

Reply:
I started going down this path but we would need to update other functions and terminology and it felt like too much (all the validate stuff). For now just leaving it as `id` and the function name calls out what kind of id.